### DISCUSSION TITLE
Problem when tring to edit Client or add new Client

### DISCUSSION THREAD

**Discussion by wajaRick** [chadwpalm/Lumunarr#29] [General]
Noticed that my lights were no longer working and couldn't access my NAS on port 3939. 
Deleted the docker container and image and created a new container running lumunarr.
Lights are now working and can access lumunarr from Firefox or Edge.
Problem: When I click on the edit icon for the Client or the 'PLUS' icon to create new Client, I get these errors:
- Issue with connection to online Plex account which requesting friends. Check logs for reason.
- Issue with connection to online Plex account while requesting account info. Check logs for reason.
- Issue with connection to online Plex account while requesting device info. Check logs for reason.
1. First minor issue on first error - "which" should be "while".
2. Looking at logs:
[2024-07-05 06:36:22] [INFO]  Retrieving Rooms
[2024-07-05 06:36:22] [INFO]  Retrieving Zones
[2024-07-05 06:36:22] [INFO]  Retrieving Light Scenes
[2024-07-05 06:36:22] [INFO]  Retrieving Smart Scenes
[2024-07-05 06:36:22] [ERROR] Issue with connection to online Plex account while requesting friends:  Request failed with status code 401
[2024-07-05 06:36:23] [ERROR] Issue with connection to online Plex account while requesting account info:  Request failed with status code 422
[2024-07-05 06:36:24] [ERROR] Issue with connection to online Plex account while requesting device info:  Request failed with status code 401
[7/5/2024 06:36:24] [INFO]  POST /backend/client - Status: 403

**Comment by wajaRick** [comment]
I ssh'd to my server and renamed settings.js to settings.js.bak.
Was able to create a new settings.js file running the Lumunarr app. 
BTW, the 10 second timeout for connecting the Bridge is a pain as the Bridge is in another part of the home.
The new settings.js does not have the problem reported.
I compared the two and settings file and there is a difference in 'token', 'uuid' , and 'thumb' avatar (still the same image.)

Suggestion: There should be a way to re-enter Plex credentials and not need to re-enter all the client settings. Mine didn't change, but maybe the 'token' expired?
I also didn't check the authorized devices in Plex to see if Lumunarr was missing. I'll do this next time if the problem happens again.

**Comment by chadwpalm** [comment]
I've been able to duplicate your issue if I remove Lumunarr from the Authorized Devices in my Plex settings. Is it possible that the device was somehow accidentally unauthorized at some point? I am not sure if Plex tokens expire over time, but I could possibly add a check when Lumunarr accesses your account and if it fails to have it reauthorize. Someone on Reddit says that tokens could become invalid if you change your Plex password or choose to sign out of all devices. Did you do either of those things?

During the login process, Lumunarr will check if the Plex token exists or is valid, and if not, it will create and store the new one in the settings file. That is why it was different in your new settings file. The UUID is unique and gets created anytime the settings file is created (usually at first use or if the settings file is deleted). The UUID is used by Plex as a unique ID for each authorized device that is connected.

> Suggestion: There should be a way to re-enter Plex credentials and not need to re-enter all the client settings. Mine didn't change, but maybe the 'token' expired?

You shouldn't have to delete or recreate your settings file when this occurs. For me, to fix the problem I simply signed out of my account using the menu on the top right in Lumunarr, and signed back in again. This creates a new Authenticated Device in your Plex account and should work again.

As far as to-dos, I can lengthen the timeout for the Bridge to say.....30 seconds? Would that be enough time? I will also look into having the app automatically sign out if the Plex token becomes invalid so you can sign in again and claim a new one.

**Reply by wajaRick** [reply]
Thanks for the quick reply. I did not change the password or sign out of all devices. I did do a few invites to share my server, but don't suspect that would have caused an issue.
Knowing that signing out and signing back in would have fixed the issue would have helped. Maybe add this to the FAQ?
30 seconds would be enough time. At 10 seconds, I can usually run and make it to the bridge. That is if my puppies are not trying to intercept me along the way. 
A catch on the 422 that signs out of Plex sound worth trying.
Thanks again.

**Comment by chadwpalm** [comment]
I've pushed an update to the development branch that addresses these issues.

> Knowing that signing out and signing back in would have fixed the issue would have helped. Maybe add this to the FAQ?

There is no need at this point since it will automatically sign out if Plex returns an unauthorized status and will require you to sign back in to generate a new token.

> 30 seconds would be enough time. At 10 seconds, I can usually run and make it to the bridge. That is if my puppies are not trying to intercept me along the way.

I went ahead and increased the timeout to 30 seconds. Hopefully that should be enough time.
